lib/widgets/admin_switch.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/material.dart';
+
+class SwitchButton extends StatefulWidget {
+  final String itemId;
+  final String categoryItemId;
+
+  const SwitchButton(
+      {super.key, required this.itemId, required this.categoryItemId});
+
+  @override
+  State<SwitchButton> createState() => _SwitchButtonState();
+}
+
+class _SwitchButtonState extends State<SwitchButton> {
+  bool light1 = true;
+
+  Future<List<String>> getPaths(String key) async {
+    List<String> paths = [];
+
+    QuerySnapshot querySnapshot = await FirebaseFirestore.instance
+        .collection('categoryItems')
+        .where('`your_field_name`', isEqualTo: key)
+        .get();
+
+    for (var doc in querySnapshot.docs) {
+      paths.add(doc.reference.path);
+    }
+
+    return paths;
+  }
+
+  Future<void> switchBooleanValue(
+    String itemKey,
+    String categoryItemKey,
+  ) async {
+    final DocumentReference itemRef =
+        FirebaseFirestore.instance.collection("items").doc(itemKey);
+
+    final DocumentSnapshot documentSnapshot = await itemRef.get();
+    final Map<String, dynamic> data =
+        documentSnapshot.data() as Map<String, dynamic>;
+
+    final bool? currentValue = data["is_available"];
+
+    if (currentValue != null) {
+      // Items collection update
+      FirebaseFirestore.instance
+          .collection("items")
+          .doc(itemKey)
+          .update({"is_available": !currentValue});
+      // Multi path update in categoryItems collection
+      FirebaseFirestore.instance.collection("categoryItems").get().then(
+        (querySnapshot) {
+          querySnapshot.docs.forEach((doc) {
+            FirebaseFirestore.instance
+                .collection("categoryItems")
```

Comment:
For me, the switch only updates the "item" but not the "categoryItem". I get a permission denied error which usually means that the query isn't well formulated. In my branch I do this query as `.collection("categoryItems/$doc.id/items").where...` and it works, so you might want to try it out.

Or the error might be in the `FirebaseFirestore.instance.collection("categoryItems").get().then` because maybe the database rules don't allow it